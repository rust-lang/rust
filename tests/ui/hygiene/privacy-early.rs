//@ edition:2018

#![feature(decl_macro)]

mod foo {
    fn f() {}
    macro f() {}

    pub macro m() {
        use f as g; //~ ERROR `f` is private, and cannot be re-exported
        f!();
    }
}

fn main() {
    foo::m!();
}
