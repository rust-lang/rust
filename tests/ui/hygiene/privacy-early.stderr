error[E0364]: `f` is private, and cannot be re-exported
  --> $DIR/privacy-early.rs:10:13
   |
LL |         use f as g;
   |             ^^^^^^
...
LL |     foo::m!();
   |     --------- in this macro invocation
   |
note: consider marking `f` as `pub` in the imported module
  --> $DIR/privacy-early.rs:10:13
   |
LL |         use f as g;
   |             ^^^^^^
...
LL |     foo::m!();
   |     --------- in this macro invocation
   = note: this error originates in the macro `foo::m` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0364`.
